proj/src/main/webapp/chat.js

```diff
@@ -236,24 +231,27 @@ async function setUserTags(tagList){
         });
 
         var currentTags = await getExistingTags(allTagsRef);
+        var currentTagRemovalDict = await getExistingTags(tagRemovalRef);
         var allTags = {};
-        keyIdDict = await getExistingTags(tagRemovalRef);
+        var tagRemovalDict = {};
 
         new Promise(async function(resolve){
             // If tag is not in existing list of tags, adds user to the chat
             // If user already has tag, deletes tag from list of current tags (as every remaining
                 // tag will be removed)
             for(var ii = 0; ii < tagList.length; ii++){
                 var tag = tagList[ii];
-                var key;
-                if(!currentTags.hasOwnProperty(tagList[ii]) && !allTags.hasOwnProperty(tagList[ii])){
-                    key = await createOrJoinChat(tag, lat, long);
+                if(!currentTags.hasOwnProperty(tag) && !allTags.hasOwnProperty(tag)){
+                    let keys = await createOrJoinChat(tag, lat, long);
+                    allTags[tag] = keys['tag'];
+                    tagRemovalDict[tag] = keys['tagRemoval'];
                 }
                 else{
-                    key = currentTags[tag];
+                    allTags[tag] = currentTags[tag];
+                    tagRemovalDict[tag] = currentTagRemovalDict[tag];
                     delete currentTags[tag];
+                    delete currentTagRemovalDict[tag];
                 }
-                allTags[tag] = key;
             }
```

Comment:
This looks like it should work, but it's pretty convoluted still. I'm good with it as long as it's tested and confirmed to work.